### PR TITLE
Implement PadInit

### PR DESCRIPTION
**Pull request by Xeeynamo**
Tested with a dual sense

### CODE REVIEW

File: src/pc/sdl2.c [hunk 1]
```diff
@@ -164,6 +169,35 @@ u_long MyPadRead(int id) {
             pressed |= PAD_R1;
         }
 
+        if (joystick) {
+            SDL_JoystickUpdate();
+            pressed |= SDL_JoystickGetButton(joystick, 0) ? PAD_CROSS : 0;
```
Comment (sozud) [inline src/pc/sdl2.c:174]: I think you're supposed to use these?
https://github.com/davidsiaw/SDL2/blob/6ecaa6b61372e5b2f9bd01201814d07e34bb4186/include/SDL_gamecontroller.h#L311

File: src/pc/sdl2.c [hunk 2]
```diff
@@ -164,6 +169,35 @@ u_long MyPadRead(int id) {
...
+            pressed |= SDL_JoystickGetButton(joystick, 2) ? PAD_TRIANGLE : 0;
+            pressed |= SDL_JoystickGetButton(joystick, 3) ? PAD_SQUARE : 0;
+            pressed |= SDL_JoystickGetButton(joystick, 4) ? PAD_L1 : 0;
+            pressed |= SDL_JoystickGetButton(joystick, 5) ? PAD_R1 : 0;
+            pressed |= SDL_JoystickGetButton(joystick, 6) ? PAD_L2 : 0;
+            pressed |= SDL_JoystickGetButton(joystick, 7) ? PAD_R2 : 0;
+            pressed |= SDL_JoystickGetButton(joystick, 8) ? PAD_SELECT : 0;
+            pressed |= SDL_JoystickGetButton(joystick, 9) ? PAD_START : 0;
+
+            short x_axis = SDL_JoystickGetAxis(joystick, 0);
```
Comment (sozud) [inline src/pc/sdl2.c:185]: https://github.com/davidsiaw/SDL2/blob/6ecaa6b61372e5b2f9bd01201814d07e34bb4186/include/SDL_gamecontroller.h#L265

**Review by sozud — APPROVED** [review]

File: src/pc/sdl2.c (hunk 1 above)
Comment (Xeeynamo) [reply src/pc/sdl2.c:174]: Weird, they do not match? Only the first four fields A, B, X and Y matches. Everything else is off. Maybe it's my configuration only. Do you have any joystick you can test this with?

Comment (sozud) [reply src/pc/sdl2.c:174]: I didn't test this, I just meant it seems like we should use the enum rather than magic numbers?

Comment (sozud) [reply src/pc/sdl2.c:174]: Reading this it seems like the better API is SDL_GameControllerGetButton?  The enums are for that and not Joystick. https://discourse.libsdl.org/t/difference-between-joysticks-and-game-controllers/24028

**Review by sozud — APPROVED** [review]

**Pull request merged**